src/jira.js

```diff
@@ -735,18 +735,39 @@ export default class JiraApi {
     }));
   }
 
+  /** Update Jira component
+   * [Jira Doc](http://docs.atlassian.com/jira/REST/latest/#api/2/component-updateComponent)
+   * @name updateComponent
+   * @function
+   * @param {string} componentId - the Id of the component to update
+   * @param {object} component - Properly Formatted Component
+   */
+  updateComponent(componentId, component) {
+    return this.doRequest(this.makeRequestHeader(this.makeUri({
+      pathname: `/component/${componentId}`,
+    }), {
+      method: 'PUT',
+      followAllRedirects: true,
+      body: component,
+    }));
+  }
+
   /** Delete component from Jira
-   * [Jira Doc](http://docs.atlassian.com/jira/REST/latest/#id290791)
+   * [Jira Doc](https://developer.atlassian.com/cloud/jira/platform/rest/v2/#api-api-2-component-id-delete)
    * @name deleteComponent
    * @function
-   * @param {string} componentId - the Id of the component to delete
+   * @param {string} id - The ID of the component.
+   * @param {string} moveIssuesTo - The ID of the component to replace the deleted component.
+   *                                If this value is null no replacement is made.
    */
-  deleteComponent(componentId) {
+  deleteComponent(id, moveIssuesTo = null) {
```

Comment:
The default value doesn't seem necessary, as well as declaring `query` in the next line. For me it would look simpler if we pass the param as following:

```js
{
  method: 'DELETE',
  followAllRedirects: true,
  qs: { moveIssuesTo },
}
```

Assuming that `request` doesn't append the querystring when `{ moveIssuesTo: undefined }`

Reply:
Looks much simpler indeed.